lib/logging/logging.go

```diff
@@ -64,6 +64,16 @@ func init() {
 	changeOwnerOfFileFunc = changeOwnerOfFile
 }
 
+type logWriter struct {
+	io.Writer
+	timeformat string
+}
+
+// custom Write to add timestamps with custom format
+func (writer logWriter) Write(bytes []byte) (int, error) {
+	return fmt.Fprintf(writer.Writer, "%s %s", time.Now().Format(writer.timeformat), string(bytes))
```

Comment:
I was of the opinion that we should avoid unnecessary memory allocation and memory copying solely for the sake of the timestamp format.
However, if we were to implement it, I think the following implementation would be better to reduce memory allocation.
```
    timestamp := time.Now().Format(writer.timeformat) + " "
    n1, err := io.WriteString(writer.Writer, timestamp)
    if err != nil {
        return n1, err
    }
    n2, err := writer.Writer.Write(bytes)
    return n1 + n2, err
```
Also, I recommend using `w` instead of `writer` for the receiver's name.